docs/source/tutorials/datasets.rst

```diff
@@ -208,9 +378,6 @@ adjustments for chosen and rejected samples in preference data.
         np.where(r_masks, CROSS_ENTROPY_IGNORE_IDX, rejected_input_ids)
     )
 
-If any of the existing dataset classes do not serve your purposes, you can similarly
-use one of them as a starting point and add the functionality you need.
-
 To be able to use your custom dataset from the config, you will need to create
```

Comment:
This is incorrect, no? As long as they fit into `chat_dataset` or `instruct_dataset`, they should be able to specify from the config without adding their own class?

Reply:
Even if it's not an Instruct dataset, I'd believe you could directly instantiate your custom dataset class without a builder.

Reply:
Let me clarify this. This really applies to the case of `InstructDataset` and `ChatDataset` which use nested components which is difficult to specify from the config. Hence, the builder functions. I made the assumption that any new dataset class would follow the same pattern.